.github/workflows/gh-ci.yaml

```
@@ -0,0 +1,279 @@
+    env:
+      CYTHON_TRACE_NOGIL: 1
+      MPLBACKEND: agg
+      GH_OS: ${{ matrix.os }}
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: setup_osx
+      if: startsWith(matrix.os, 'macOS')
+      run: |
+        # Set OS specific vars and compiler flags
+        echo "OS_NAME=osx" >> $GITHUB_ENV
+        # TODO: work out why this is necessary (from CI helpers)
+        echo "MACOSX_DEPLOYMENT_TARGET=10.9" >> $GITHUB_ENV
+        ulimit -S -n 2048
+        clang -v
+        echo "CC=clang" >> $GITHUB_ENV
+        clang++ -v
+        echo "CXX=clang++" >> $GITHUB_ENV
+        gfortran-9 -v
+        echo "FC=gfortran-9" >> $GITHUB_ENV
+
+    - name: setup_linux
+      if: startsWith(matrix.os, 'ubuntu')
+      run: |
+        # Set OS specific vars and compiler flags
+        echo "OS_NAME=linux" >> $GITHUB_ENV
+        gcc -v
+        echo "CC=gcc" >> $GITHUB_ENV
+        g++ -v
+        echo "CXX=g++" >> $GITHUB_ENV
+        gfortran -v
+        echo "FC=gfortran" >> $GITHUB_ENV
+
+    - name: setup_miniconda
+      uses: conda-incubator/setup-miniconda@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+        auto-update-conda: true
+        channel-priority: flexible
+        channels: biobuilds, conda-forge
+        add-pip-as-python-dependency: true
+        # TODO: mamba causes pip to segfault, switch when fixed
+        #mamba-version: "*"
+        architecture: x64
+        
+    - name: install_deps
+      env:
+        MDA_CONDA_FULL_DEPS: "${{ env.MDA_CONDA_MIN_DEPS }} ${{ env.MDA_CONDA_EXTRA_DEPS }}"
+        MDA_PIP_FULL_DEPS: "${{ env.MDA_PIP_MIN_DEPS }} ${{ env.MDA_PIP_EXTRA_DEPS }}"
+      run: |
+        # NOTE: vars need to be re-assigned
+        # NOTE: set matrix.numpy to pin to a specific numpy version
+        conda_deps="${{ matrix.numpy }} ${MDA_CONDA_${{ matrix.run_type }}_DEPS}"
+        pip_deps=${MDA_PIP_${{ matrix.run_type }}_DEPS}
+        conda install ${conda_deps}
+        pip install ${pip_deps}
+
+        # NOTE: set setuptools version if running asv test
+        if [ ${{ matrix.name }} = "asv_check" ]; then
+          pip install asv "setuptools<45.0.0"
+        fi
+    
+    - name: check_setup
+      run: |
+        # Check OS and python setup
+        echo "OS: ${OS_NAME}"
+        which python
+        which pip
+        pip list
+        conda info
+        conda list
+
+    - name: install_hole
+      if : matrix.install_hole
+      run: |
+        # We manually build hole2 to avoid OS incompatibilities
+        git clone https://github.com/MDAnalysis/hole2.git
+        cd hole2/src
+        source ../source.apache
+        (make FC=${FC}) && (make PREFIX=${HOME}/hole2 FC=${FC} install)
+        source ../source.unset
+        echo "HOLE_BINDIR=${PWD}/../exe" >> $GITHUB_ENV
+        echo "${PWD}/../exe" >> $GITHUB_PATH
+
+    - name: install_mda
+      run: |
+        (cd package/ && python setup.py develop) && (cd testsuite/ && python setup.py install)
```

How about doing a proper install instead of develop?
```suggestion
        (cd package/ && python setup.py install) && (cd testsuite/ && python setup.py install)
```
This can catch the occasional mistake in missing files.

---

I was doing this until: https://github.com/MDAnalysis/mdanalysis/pull/3040#issuecomment-734985358

The issue is that doing an `install` results in a lack of coverage for the `.pyx` files. I'll see if doing an `--inplace` call might work here.

---

With further reading, I don't think that'll work. If someone has an idea here I can implement it, otherwise I suggest we go with the `develop` setup for now?

---

Hm, ok, I bow to facts. Perhaps add a comment to the file as to why we're doing develop install, just so that future me is not asking the same question again?

---

I have a feeling it is fixeable, but it's probably one of those obvious things I can't find in the setuptools docs (I really need to sit down and learn it all properly one day) 🙃 I'll tag it as `TODO`, I do agree that using `install` everywhere would be much better.